### PR TITLE
Make response headers as optional

### PR DESCRIPTION
**Pull request by raych1**
It's base on swagger 2.0 specification.

### CODE REVIEW

**Review by ruowan — APPROVED** [review]

File: lib/liveValidation/operationValidator.ts [hunk 1]
```diff
@@ -358,21 +357,11 @@ const validateLroHeader = (
...
-      (properties.provisioningState === undefined ||
-        properties.provisioningState === "Succeeded" ||
-        properties.provisioningState === "Failed" ||
-        properties.provisioningState === "Canceled")
-    ) {
-      // Ignore LRO header check when it's sync call
-      return;
-    }
+  if (statusCode === "201") {
+    // Ignore LRO header check cause RPC says azure-AsyncOperation is optional if using 201/200+ provisioningState
```
Comment (ruowan) [inline lib/liveValidation/operationValidator.ts:363]: What if statusCode is  202? Do we validate 202 header?

File: lib/liveValidation/operationValidator.ts [hunk 2]
```diff
@@ -322,31 +322,30 @@ const validateLroOperation = (
   operation: Operation,
   statusCode: string,
   headers: StringMap<string>,
-  body: any,
```
Comment (ruowan) [inline lib/liveValidation/operationValidator.ts:325]: Is body unused?

File: lib/liveValidation/operationValidator.ts (hunk 1 above)
Comment (raych1) [reply lib/liveValidation/operationValidator.ts:363]: Yes, 202 case has to provide LRO headers.

File: lib/liveValidation/operationValidator.ts (hunk 2 above)
Comment (raych1) [reply lib/liveValidation/operationValidator.ts:325]: Yes, no need any more.

**Pull request merged**